experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -96,21 +97,34 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	 */
 	public constructor(
 		private head: GraphCommit<TChange>,
-		private readonly sessionId: string,
+		public readonly sessionId: string,
 		public readonly changeFamily: ChangeFamily<TEditor, TChange>,
 		public undoRedoManager: UndoRedoManager<TChange, TEditor>,
 		private readonly anchors?: AnchorSet,
 	) {
 		super();
 		this.editor = this.changeFamily.buildEditor(
-			(change) => this.applyChange(change, UndoRedoManagerCommitType.Undoable),
+			(change) =>
+				this.applyChange(change, mintRevisionTag(), UndoRedoManagerCommitType.Undoable),
 			new AnchorSet(), // This branch class handles the anchor rebasing, so we don't want the editor to do any rebasing; so pass it a dummy anchor set.
 		);
 	}
 
-	private applyChange(change: TChange, undoRedoType: UndoRedoManagerCommitType): void {
+	/**
+	 * Sets the head of this branch. Emits no change events.
+	 */
+	public setHead(head: GraphCommit<TChange>): void {
```

Comment:
I'm not super happy about this, but we need it for when the edit manager loads from a summary and needs to "reset" the local branch, but without "mutating" it and emitting change events. I'm hopeful that in future refactors, the ownership of the local branch will change and maybe a better pattern will emerge.